### DISCUSSION TITLE
Refresh shown windows more quickly when switching to another space

### DISCUSSION THREAD

**Discussion by andreyrd** [lawand-dot-io/taskbar#56] [Feature requests]
I don't know exactly how you are rendering a taskbar on each space, so this may not actually be technically possible to fix. But right now switching between Spaces, it takes a second for the window list to update to the Space we just switched to.

From my experience implementing this kind of app - the notification you get for a space switch is very delayed, at the end of the animation and then some. But if each space had it's own render of the taskbar, that could be a workaround. And then each one could maintain a cache of what windows were open on that space.

**Comment by lawand-dot-io** [comment]
Hello. Thanks for the feature request. From what I know, macOS doesn't give access to the windows that are on other spaces, so I need to wait until the space move animation finishes (as you mentioned) and then I get the list of windows on the space we just moved to, so I replace the current ones with the ones from the new space. And by the way I render one Taskbar, not one per space :) It just stays shown whenever you move into a different space. In any case I will keep this feature request open and if it received more and more upvotes I could do more research on this. Cheers

**Reply by andreyrd** [reply]
Understood. I'm doing similar in my own. One idea I've considered is to keep a cached list of the windows on a space, and show that when the switch begins, then update it when it ends. But unfortunately, I don't know of any way to get a notification that the space switch is starting that doesn't involve hacks like keeping invisible full screen windows on each space (I believe this is what Swindler does but I may be misremembering a different project) or screen recording (which is maybe the worst way).

**Reply by elo-siema** [reply]
@lawand-dot-io Could you cache the open windows on non-active spaces? It's unlikely this set would have changed while a space is inactive in general usage. Perceived latency would go way down.

**Reply by lawand-dot-io** [reply]
@elo-siema yeah, that could be one way to do it. I could test this out and see whether or not it's buggy. But to be honest in general this is not too much of a high priority right now, so perhaps I will get a chance to look into this in the future. Thanks
